human-tracking-and-pose-detection/main.py

```diff
@@ -1,234 +1,144 @@
-    # Invoke inference.
-    interpreter.invoke()
-    # Get the model prediction.
-    keypoints_with_scores = interpreter.get_tensor(output_details[0]['index'])
-    return keypoints_with_scores[0][0]
-elif "movenet_multipose" in model_name:
-  module = hub.load("https://tfhub.dev/google/movenet/multipose/lightning/1")
-  input_size = 256
-  def movenet(input_image):
-    model = module.signatures['serving_default']
-
-    input_image = tf.cast(input_image, dtype=tf.int32)    # SavedModel format expects tensor type of int32.
-    outputs = model(input_image)
-    # Output is a [1, 6, 56] tensor.
-    keypoints_with_scores = outputs['output_0'].numpy()
-    threshold =0.3
-    count_of_people = np.sum(keypoints_with_scores[0,:,-1]>threshold)
-    print("count_of_people",count_of_people)
-    return keypoints_with_scores[:,:,:51].reshape((6,17,3))[0]
-else:
-  if "movenet_lightning" in model_name:
-    if downloadModel:  
-      module = hub.load("https://tfhub.dev/google/movenet/singlepose/lightning/4")
-    input_size = 192
-  elif "movenet_thunder" in model_name:
-    module = hub.load("https://tfhub.dev/google/movenet/singlepose/thunder/4")
-    input_size = 256
-  else:
-    raise ValueError("Unsupported model name: %s" % model_name)
-
-  def movenet(input_image):
-    """Runs detection on an input image.
-
-    Args:
-      input_image: A [1, height, width, 3] tensor represents the input image
-        pixels. Note that the height/width should already be resized and match the
-        expected input resolution of the model before passing into this function.
-
-    Returns:
-      A [1, 1, 17, 3] float numpy array representing the predicted keypoint
-      coordinates and scores.
-    """
-    model = module.signatures['serving_default']
-
-    input_image = tf.cast(input_image, dtype=tf.int32)    # SavedModel format expects tensor type of int32.
-    outputs = model(input_image)
-    # Output is a [1, 1, 17, 3] tensor.
-    keypoints_with_scores = outputs['output_0'].numpy()
-    return keypoints_with_scores[0][0]
-
-
+import time, math
+
+if parser_args.display_image:
+  from helper_for_visualization import draw_prediction_on_image
+
+#print only if verbose==True
+def print_verbose_only(*args, **kwargs):
+  verbose = kwargs.pop('verbose', False)
+  if parser_args.verbose: print(*args, **kwargs)
+
+#function decorator for printing fps
+def print_fps(func):
+  def wrapper(*args, **kwargs):
+    prev_time = time.time()
+    result = func(*args, **kwargs)
+    cur_time = time.time()
+    if cur_time - prev_time > (0.000001):
+      fps = 1.0 / (cur_time - prev_time)
+      print_verbose_only('FPS: {:.2f}'.format(fps))
+    return result
+  return wrapper
+
+#movenet model. Gets input image and outputs array of keypoints with certainty score
+#downloaded from #https://tfhub.dev/google/movenet/multipose/lightning/1
+model = hub.load("models_downloaded/multipose_tf").signatures['serving_default'] 
```

Comment:
Load the model in main() and pass it as parameter to other functions using it

Reply:
solved